lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -483,11 +491,22 @@ private synchronized ErrorInfo waitForChange() {
          * ConnectionStateListener interface
          */
         @Override
-        public void onConnectionStateChanged(ConnectionStateChange state) {
-            synchronized(this) {
-                change = state;
-                notify();
+        public synchronized void onConnectionStateChanged(ConnectionStateChange state) {
+            change = state;
+            notify();
+        }
+
+        /**
+         * Remove this ConnectionWaiter as a connection listener.
+         */
+        private void close() {
+            // This method is explicitly not synchronized. There may be a case for this in the
```

Comment:
In this repo we've tended to use `/* .... */` comment style

Reply:
For interface commentary I would agree, however within a method implementation I tend to prefer `//` as it's more compact - typically for single line comments but also for multi-line. I should document this somewhere.